node/impl/full/actor_events.go

```diff
@@ -195,54 +237,41 @@ func (a *ActorEventHandler) SubscribeActorEvents(ctx context.Context, evtFilter
 			// tell the caller we're done
 			close(out)
 			fm.ClearSubChannel()
-			if err := a.EventFilterManager.Remove(ctx, fm.ID()); err != nil {
+			if err := a.eventFilterManager.Remove(ctx, fm.ID()); err != nil {
 				log.Warnf("failed to remove filter: %s", err)
 			}
 		}()
 
+		// When we start sending real-time events, we want to make sure that we don't fall behind more
+		// than one epoch's worth of events (approximately). Capture this value now, before we send
+		// historical events to allow for a little bit of slack in the historical event sending.
+		minBacklogHeight := a.chain.GetHeaviestTipSet().Height() - 1
```

Comment:
it's not `panic` racy though, right? it just needs to be approximate, we need an end condition and if it ends up being 2, or even 3 epochs behind rather than 1 then that's OK, as long as there's a termination condition

Reply:
Discussing it below.